frontend/controller/serviceworkers/sw-primary.js

```diff
@@ -265,6 +267,11 @@ sbp('okTurtles.events/on', CHELONIA_RESET, () => {
   sbp('gi.periodicNotifications/init')
 })
 
+let currentVersionInfo
+sbp('okTurtles.events/on', NOTIFICATION_TYPE.VERSION_INFO, (versionInfo) => {
+  currentVersionInfo = versionInfo
+})
+
```

Comment:
Where is this event being emitted from? The web socket?

How do we know this will happen before the `ready` event?


(If this really needs to be here and be done this way, then a comment is definitely needed explaining all of this. Otherwise should be modified + probably a comment anyway)

Reply:
The web socket cannot emit events, but, yes, the WS receives a message that results in an event being emitted if there's a version discrepancy. This event is emitted soon after connection.

> How do we know this will happen before the ready event?

We don't.

Reply:
> We don't.

Then what is the point?

We need to immediately do the version check when the pubsub emits the event and then tell the browser to do the page refresh.

If this only happens on the `ready` event it defeats the purpose of this PR. Unless I'm missing something.

Reply:
> We need to immediately do the version check when the pubsub emits the event and then tell the browser to do the page refresh.

No, we don't. We need to send the event when it's available, which is what this PR does. This doesn't only happen on the `ready` event.

Reply:
> On already connected clients, when the event is received.

Can you point to where this happens?

If I'm using GI right now (client open to sw), and an update happens while I'm using it, and the pubsub sends the event to the sw, where is the logic that triggers the page refresh immediately?

Reply:
> If I'm using GI right now (client open to sw), and an update happens while I'm using it, and the pubsub sends the event to the sw, where is the logic that triggers the page refresh immediately?

There isn't. That's why I said, 'with this PR'.

> Can you point to where this happens?

As mentioned, the event is emitted on (re)connection to the WS. This event is forwarded from the SW to the browser. In addition, the event is sent to new clients when they connect to the SW.

Reply:
> There isn't. That's why I said, 'with this PR'.

And I'm asking, where in this PR, does the websocket trigger the page refresh immediately in connected clients?

Reply:
In `frontend/controller/backend.js`